tests/test_graphBuilder.py

```diff
@@ -0,0 +1,44 @@
+# This file is part of pipe_base.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Tests of things related to the GraphBuilder class."""
+
+import unittest
+
+from lsst.pipe.base import GraphBuilder
+import lsst.utils.tests
+
+
+class HelperTestCase(unittest.TestCase):
+
+    def test_getInstrument(self):
+        """Test getting the instrument from the query."""
+        queries = (("tract = 42 and instrument = 'HSC'", "HSC"),
+                   ("tract=42 and INSTRUMENT='HSC'", "HSC"),
+                   ('tract=42 and Instrument =  "HSC"', "HSC"),
```

Comment:
There need to be tests where instrument is first and then tract is second, and also one where instrument is in the middle of two other clauses.

Reply:
and I guess also a `tract=42 and notinstrument = "HSC" and instrument = "LSSTCam"`